model/riscv.sail_project

```diff
@@ -0,0 +1,365 @@
+        riscv_insts_vext_fp_utils.sail,
+        riscv_insts_vext_vset.sail,
+        riscv_insts_vext_arith.sail,
+        riscv_insts_vext_fp.sail,
+        riscv_insts_vext_mem.sail,
+        riscv_insts_vext_mask.sail,
+        riscv_insts_vext_vm.sail,
+        riscv_insts_vext_fp_vm.sail,
+        riscv_insts_vext_red.sail,
+        riscv_insts_vext_fp_red.sail,
+     }
+  }
+
+  // RISC-V Cryptography Extension
+  K {
+    requires riscv_core
+
+    K_core {
+      files riscv_types_kext.sail
+    }
+    Zkn {
+      requires riscv, K_core
+      files riscv_insts_zkn.sail
+    }
+    Zks {
+      requires riscv, K_core
+      files riscv_insts_zks.sail
+    }
+    Zkr {
+      requires riscv, K_core
+      files riscv_zkr_control.sail
+    }
+    Zbkb {
+      requires riscv
+      files riscv_insts_zbkb.sail
+    }
+    Zbkx {
+      requires riscv
+      files riscv_insts_zbkx.sail
+    }
+  }
+
+  vector_crypto {
+    requires riscv_core
+
+    Zvk_core {
+      requires V_core, K_core
+      before riscv
+      files riscv_zvk_utils.sail
+    }
+
+    Zvbb {
+      requires riscv, V, K, Zvk_core
+      files riscv_insts_zvbb.sail
+    }
+    Zvbc {
+      requires riscv, V, K, Zvk_core
+      files riscv_insts_zvbc.sail
+    }
+    Zvkg {
+      requires riscv, V, K, Zvk_core
+      files riscv_insts_zvkg.sail
+    }
+    Zvkned {
+      requires riscv, V, K, Zvk_core
+      files riscv_insts_zvkned.sail
+    }
+    Zvksed {
+      requires riscv, V, K, Zvk_core
+      files riscv_insts_zvksed.sail
+    }
+    Zvknhab {
+      requires riscv, V, K, Zvk_core
+      files riscv_insts_zvknhab.sail
+    }
+    Zvksh {
+      requires riscv, V, K, Zvk_core
+      files riscv_insts_zvksh.sail
+    }
+  }
+
+  // Control and Status Register (CSR) Instructions
+  Zicsr {
+    requires riscv_core, riscv, riscv_exceptions, pmp, V_core
+    files riscv_insts_zicsr.sail
+  }
+
+  Svinval {
+    requires riscv_core, riscv, I
+    files riscv_insts_svinval.sail
+  }
+
+  Zihpm {
+    requires riscv_core, riscv
+    files riscv_zihpm.sail
+  }
+
+  Smcntrpmf {
+    requires riscv_core
+    files riscv_smcntrpmf.sail
+  }
+
+  Sscofpmf {
+    requires riscv_core, riscv, Zihpm
+    files riscv_sscofpmf.sail
+  }
+
+  Sstc {
+    requires riscv_core, riscv, Zihpm
+    files riscv_sstc.sail
+  }
+
+  Zawrs {
+    requires riscv_core, riscv
+    files riscv_insts_zawrs.sail
+  }
+
+  Zicond {
+    requires riscv_core, riscv
+    files riscv_insts_zicond.sail
+  }
+
+  Zicntr {
+    requires riscv_core, riscv
+    files riscv_zicntr_control.sail
+  }
+
+  Zicbom {
+    requires riscv_core, riscv
+    files riscv_insts_zicbom.sail
+  }
+
+  Zicboz {
+    requires riscv_core, riscv
+    files riscv_insts_zicboz.sail
+  }
+
+  Zifenci {
+    requires riscv_core, riscv
+    files riscv_insts_zifencei.sail
+  }
+}
+
+// May-be-operations (MOPS) defined after extensions so they can be
+// overriden by earlier extensions
+mops {
+  after extensions
+
```

Comment:
Could have `requires riscv_core, riscv` here since it is common to both `mop` extensions.

Reply:
Done